lib/output_xunit.c

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (C) 2021  Red Hat, Inc.
+ * Author(s):  David Cantrell <dcantrell@redhat.com>
+ *
+ * This program is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program.  If not, see
+ * <https://www.gnu.org/licenses/>.
+ *
+ * SPDX-License-Identifier: LGPL-3.0-or-later
+ */
+
+#include <stdio.h>
+#include <err.h>
+#include <assert.h>
+
+#include "rpminspect.h"
+
+/*
+ * Output a results_t in XUnit format.  This can be consumed by Jenkins or
+ * other services that can read in XUnit data (e.g., GitHub).
+ */
+void output_xunit(const results_t *results, const char *dest, const severity_t threshold) {
+    results_entry_t *result = NULL;
+    int r = 0;
+    int count = 0;
+    int total = 0;
+    int failures = 0;
+    FILE *fp = NULL;
+    const char *header = NULL;
+
+    /* count up total test cases and total failures */
+    TAILQ_FOREACH(result, results, items) {
+        if (header == NULL || strcmp(header, result->header)) {
+            total++;
+        }
+
+        header = result->header;
+    }
+
+    /* default to stdout unless a filename was specified */
+    if (dest == NULL) {
+        fp = stdout;
+    } else {
+        fp = fopen(dest, "w");
+
+        if (fp == NULL) {
+            warn(_("opening %s for writing"), dest);
+            return;
+        }
+    }
+
+    header = NULL;
+    fprintf(fp, "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n");
+    fprintf(fp, "<testsuite tests=\"%d\" failures=\"%d\" errors=\"0\" skipped=\"0\">\n", total, failures);
```

Comment:
Oh, nice -- I was afraid to open this pull request because I thought there will be some libxml2 madness in here :)

Nicely done David :wink: